### ISSUE TITLE
Couldn't resolve dependency iron-autogrow-textarea@PolymerElements/iron-autogrow-textarea#^1.0.0 for PolymerElements/paper-input

### ISSUE THREAD

**Issue by jsundqvist** [craigburke/client-dependencies-gradle#43]
Hi Craig, thanks for an awesome plugin.

I'm trying to include `PolymerElements/paper-input` from GitHub as a bower module but the `clientInstall` task fails on a dependency.

My build.gradle contains:

```
clientDependencies {
    bower {
        'PolymerElements/paper-input'('1.0.0') { source '*' }
    }
}
```

Upon running `clientInstall` I get the following error:

```
Execution failed for task ':clientInstall'.
> com.craigburke.gradle.client.registry.core.DependencyResolveException: Couldn't resolve iron-autogrow-textarea@PolymerElements/iron-autogrow-textarea#^1.0.0
```

The `bower.json` for `PolymerElements/paper-input` contains:

```
"dependencies": {
    "polymer": "Polymer/polymer#^1.2.0",
    "iron-autogrow-textarea": "PolymerElements/iron-autogrow-textarea#^1.0.0", // <--
    "iron-behaviors": "PolymerElements/iron-behaviors#^1.0.0",
    "iron-form-element-behavior": "PolymerElements/iron-form-element-behavior#^1.0.0",
    "iron-input": "PolymerElements/iron-input#^1.0.0",
    "paper-styles": "PolymerElements/paper-styles#^1.1.4",
    "iron-a11y-keys-behavior": "PolymerElements/iron-a11y-keys-behavior#^1.0.0"
}
```

Thankful for any help on this issue!

/Johannes

**Comment by craigburke** [comment]
@jsundqvist yeah this is helpful. I'll take a look.

**Comment by craigburke** [comment]
Should be able to parse versions in the following formats:
- `org/repo`
- `org/repo#version` 
- `org/repo#tag`
- `org/repo#sha`
- `org/repo#branch`
- `org/repo@version`

See: 
- http://bower.io/docs/api/#install
- https://docs.npmjs.com/cli/install

**Comment by jsundqvist** [comment]
Are there plans on implementing support for version ranges?
https://docs.npmjs.com/misc/semver#ranges

I believe the type in question is:
https://docs.npmjs.com/misc/semver#caret-ranges-123-025-004

Thanks again.

/Johannes

**Comment by craigburke** [comment]
@jsundqvist version ranges are already implemented, if you run into a case where they don't work let me know.

**Comment by craigburke** [comment]
See the relevent tests here: https://raw.githubusercontent.com/craigburke/client-dependencies-gradle/master/src/test/groovy/com/craigburke/gradle/client/dependency/VersionResolverSpec.groovy

**Comment by jsundqvist** [comment]
OK, my bad. The issue I have submitted is the one I'm having issues with. Are you not able to reproduce it?

**Comment by craigburke** [comment]
@jsundqvist I am able to produce it. I have tests written I just have to get them passing now :)

Hopefully I'll be able to fix it sometime tomorrow.

**Comment by craigburke** [comment]
This is resolved in version 1.1.3.
